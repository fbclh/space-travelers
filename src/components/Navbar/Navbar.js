import { NavLink } from 'react-router-dom';
import logo from '../../assets/logo.png';

const Navbar = () => {
  const Links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/profile',
      text: 'Profile',
    },
  ];

  return (
    <nav className="navbar">
      <div className="logo-container">
        <img src={logo} alt="logo" style={{ width: '50px', height: '50px' }} />
        <h1>Space Travelers</h1>
      </div>
      <ul className="ul-links">
        {Links.map((link) => (
          <li className="li-links" key={link.id}>
            <NavLink className="links" to={link.path}>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
