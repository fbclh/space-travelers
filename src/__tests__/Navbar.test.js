import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar/Navbar';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
