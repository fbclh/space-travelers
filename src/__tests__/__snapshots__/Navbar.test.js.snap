// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="logo-container"
  >
    <img
      alt="logo"
      src="logo.png"
      style={
        Object {
          "height": "50px",
          "width": "50px",
        }
      }
    />
    <h1>
      Space Travelers
    </h1>
  </div>
  <ul
    className="ul-links"
  >
    <li
      className="li-links"
    >
      <a
        aria-current="page"
        className="links active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="li-links"
    >
      <a
        className="links"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li
      className="li-links"
    >
      <a
        className="links"
        href="/profile"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
  </ul>
</nav>
`;
